chore: fixed some HTML violations

UVLabs opened the pull request:
### Summary
Role attribute is not required on `<header>` and `<main>` elements as they are assumed by default by screen readers as documented here: 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role#description
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/main_role#description

@cristian-ungureanu For the [next-page-hide](https://github.com/Codeinwp/neve/blob/master/header.php#L78) attribute added to the elements, I'm not sure why it was added, I did a project wide search of neve and neve pro and I could have only found it being applied but not being used...so I left it because I'm not sure if we're making use of this somewhere or somehow in another project 

### Will affect visual aspect of the product
<!-- It includes visual changes? -->
NO

### Screenshots <!-- if applicable -->

### Test instructions
<!-- Describe how this pull request can be tested. -->

- Realistic test would be using a screen reader, otherwise we just need to make sure there's nothing strange happening with our starter sites header and footer and AMP

<!-- Issues that this pull request closes. -->
Closes https://github.com/Codeinwp/neve-pro-addon/issues/1839
<!-- Should look like this: `Closes #1, #2, #3.` . -->

cristian-ungureanu commented:
@UVLabs the `next-page-hide` is related to AMP https://amp.dev/documentation/components/amp-next-page/. It's a marker to exclude that element when a new page is loading in infinite scroll.
Everything else looks good, thank you! 🚀

cristian-ungureanu commented:
Now that you're mentioning this, I think you're right. We can use the `neve_is_amp` function to show it just in an amp context. Thank you @selul!

cristian-ungureanu commented on header.php:
```diff
@@ -75,7 +75,7 @@
 	?>
 
-	<header class="<?php echo esc_attr( $header_classes ); ?>" role="banner" next-page-hide>
+	<header class="<?php echo esc_attr( $header_classes ); ?>" next-page-hide>
```
Let's use `neve_is_amp` function to display the `next-page-hide` just in an AMP context.

cristian-ungureanu commented on header.php:
```diff
@@ -75,7 +75,7 @@
 	?>
 
-	<header class="<?php echo esc_attr( $header_classes ); ?>" role="banner" next-page-hide>
+	<header class="<?php echo esc_attr( $header_classes ); ?>" next-page-hide>
```
Let's use `neve_is_amp` function to display `next-page-hide` just in an AMP context

UVLabs replied:
@cristian-ungureanu I've made the change here

cristian-ungureanu commented:
Thank you @UVLabs, I moved this to QA, maybe @irinelenache can rerun the checker in different contexts and see if there's something else we could do.

irinelenache commented:
@UVLabs @cristian-ungureanu Tested and found a few other errors, i don't know how relevant they are but will show them here:

- [ ] These ones appear when **Search icon** and **Search form** are added inside the header
![image](https://user-images.githubusercontent.com/67897586/163561440-ab1c50df-6efb-4397-8faa-6245a847a98b.png)

- [ ] These ones are related to menu items that contain subitems (With **Shop** starter site imported)
![image](https://user-images.githubusercontent.com/67897586/163561637-84fa99b4-050f-4d41-a037-d6abd1e6cdf5.png)
![image](https://user-images.githubusercontent.com/67897586/163561663-ca07b749-7540-4906-a78c-61a43ea08c6a.png)

- [ ] This one related to the **Scroll to top**
![image](https://user-images.githubusercontent.com/67897586/163561752-8103b5cd-d8b9-404d-9ac0-49c30ab50bcd.png)

Didn't find any problem regarding this PR, keyboard navigating works as before 👍

UVLabs commented:
@irinelenache I addressed the first issue and tested and it seems the element type doesn't matter in this case.

However, for the other two points. In point 2 we need this otherwise sidebar carret doesn't become focusable, I can see a comment was added here for this: https://github.com/Codeinwp/neve/blob/master/inc/views/nav_walker.php#L64

Regarding point 3, I think this has to do with AMP (@cristian-ungureanu please confirm) so I changed the code a bit so it outputs this markup only if amp.

Please test again including AMP and let me know if there's any regression

irinelenache commented:
@UVLabs Checked again and the **Search icon** and **Search form** errors are still there, but changed a bit
![image](https://user-images.githubusercontent.com/67897586/165468820-9e4bf197-2c64-4416-95a6-78891f20bf33.png)

Rest of the errors are still there, didn't find any regression though 👍

UVLabs commented:
@irinelenache Sorry i missed that error, it should be fine now, I made some changes to the actual element used, from a `div` to a `span` since block elements are not allowed inside `button` elements, everything seems to have remained the same but please help confirm

irinelenache commented:
@UVLabs Tested and the **Search icon** and **Search form** issues are fixed, thanks 🚀 
This can be moved to **Ready to merge** after all checks are passing 👍
